### PR TITLE
[incubator/sparkoperator] Revert sparkJobNamespace value 

### PR DESCRIPTION
**Pull request by MelnykAndriy**
Signed-off-by: Melnyk Andriy <kukoban@ukr.net>

#### What this PR does / why we need it:

This PR reverts sparkJobNamespace to the value which is documented in the docks.
Changes made with https://github.com/helm/charts/commit/90e572714376c926b72ae99c6afe7b3b441f862e
commit has led to missed creation of a `Role` and `RoleBinding` for the service account of spark driver.

It also broke [spark-on-k8s-operator Quick Start Guide](https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/master/docs/quick-start-guide.md) because 
```bash
helm install incubator/sparkoperator --namespace spark-operator
```
became semantically different from  
```bash
helm install incubator/sparkoperator --namespace spark-operator --set sparkJobNamespace=default
```

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped

### CODE REVIEW

**Comment by yuchaoran2011** [comment]
@liyinan926 Do you think we should merge this PR or update the doc to reflect that the job namespace is no longer set by default?

**Comment by liyinan926** [comment]
I actually think it largely depends on user expectation of what the Helm chart should do. If the expectation of most users is the installation should create the necessary RBAC settings for a namespace where they can run SparkApplications in, then I guess we should restore the part that creates the RBAC settings.

**Comment by yuchaoran2011** [comment]
Agreed. My preference is the old way: have the chart set up RBAC in a default namespace for me.

**Comment by liyinan926** [comment]
@yuchaoran2011 I agree.

File: incubator/sparkoperator/templates/spark-rbac.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{- if and (.Values.rbac.create) (ne .Values.sparkJobNamespace "") }}
```
Comment (liyinan926) [inline incubator/sparkoperator/templates/spark-rbac.yaml:1]: I think this change should be kept, i.e., skipping creating RBAC if the namespace value is empty.

Comment (MelnykAndriy) [reply incubator/sparkoperator/templates/spark-rbac.yaml:1]: I think It is not documented and not obvious.
In order to skip RBAC creation user will have to write.
```
helm install incubator/sparkoperator --namespace spark-operator --set sparkJobNamespace=
```
I suggest that adding a separate flag for enabling/disabling RBAC creation is better.
Any way I can revert it.
Should I do it?

Comment (yuchaoran2011) [reply incubator/sparkoperator/templates/spark-rbac.yaml:1]: Good point. There's currently an `rbac.create` in `values.yaml`. How about splitting it into `rbac.createOperator` and `rbac.createSparkJob`?

Comment (liyinan926) [reply incubator/sparkoperator/templates/spark-rbac.yaml:1]: I think we can fix the documentation instead.

Comment (yuchaoran2011) [reply incubator/sparkoperator/templates/spark-rbac.yaml:1]: That's fine with me.

Comment (liyinan926) [reply incubator/sparkoperator/templates/spark-rbac.yaml:1]: @MelnykAndriy can you update the documentation? We can get this merged after that. Thanks!

**Pull request closed without merging**